### PR TITLE
Add compile stage to travis script

### PR DESCRIPTION
**Pull request by chriseclectic**
### Summary

Following on from #553, this PR helps clean up travis build logs and test by adding an extra `compile` stage to the travis tests.
* The compile stage tests building the Python wheels on Linux and MacOS
* The test stage is modified to only run the python unittest scripts so that the full CMake build log isn't returned in the output script.
* Move Python 3.8 MacOS tests to kron job since they are taking too long to run (3-4 x slower than 3.7 tests)

This helps fix issue #548 as we add more build files and more tests files leading to longer and longer logs.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -112,10 +121,36 @@ jobs:
...
+    - stage: compile
+      name: Test Build MacOS
+      <<: *stage_osx
+      before_script: true
+      script:
+        - python setup.py bdist_wheel -- -- -j4
+    # Power Build
+    - stage: compile
+      name: Test Build Linux (Power)
+      arch: ppc64le
```
Comment (atilag) [inline .travis.yml]: ouch! Does this exist on Travis???? Didn't know... great!

File: .travis.yml [hunk 2]
```diff
@@ -86,10 +94,10 @@ stage_osx: &stage_osx
       if [ ! "$(ls -A .stestr)" ]; then
         rm -rf .stestr
       fi
-
-  script:
-    - python setup.py bdist_wheel -- -- -j4
+    # Build and install qiskit-aer package
+    - python setup.py bdist_wheel -- -DAER_THRUST_BACKEND=OMP -- -j4
```
Comment (atilag) [inline .travis.yml]: did you leave `AER_THRUST_BACKEND` on purpose?

Comment (chriseclectic) [reply .travis.yml]: Typo from gpu branch

File: .travis.yml [hunk 3]
```diff
@@ -31,6 +31,9 @@ stage_generic: &stage_generic
     - pip install cython
     # Installing qiskit-terra master branch...
     - pip install https://github.com/Qiskit/qiskit-terra/archive/master.zip
+    # Installing qiskit-ibmq-provider stable branch
+    # This is only needed to suppress warnings when importing Qiskit
+    - pip install https://github.com/Qiskit/qiskit-ibmq-provider/archive/stable.zip
```
[the remark this answers is not in the document]
Comment (chriseclectic) [reply .travis.yml]: This was just so `import qiskit` didnt print warnings

[the remark this answers is not in the document]
Comment (mtreinish) [reply .travis.yml]: There was an open terra PR to add an env variable to disable this warning. Right now, you can use https://docs.python.org/3/library/warnings.html#warnings.filterwarnings to suppress it from python (as documented at the bottom of: https://qiskit.org/documentation/contributing_to_qiskit.html#installing-terra-from-source). But having an external option to do that  via and env variable would be good for cases like this (and via a user config file for local use too).

**Pull request merged**